packages/core/src/resources/manager.ts

```diff
@@ -0,0 +1,182 @@
+import type { MCPManager } from '../mcp/manager.js';
+import type { ResourceSet, ResourceMetadata } from './types.js';
+import { InternalResourcesProvider } from './internal-provider.js';
+import type { ReadResourceResult } from '@modelcontextprotocol/sdk/types.js';
+import type { ValidatedInternalResourcesConfig } from './schemas.js';
+import { logger } from '../logger/index.js';
+import { ResourceError } from './errors.js';
+
+export interface ResourceManagerOptions {
+    internalResourcesConfig?: ValidatedInternalResourcesConfig;
+}
+
+export class ResourceManager {
+    private mcpManager: MCPManager;
+    private internalResourcesProvider?: InternalResourcesProvider;
+
+    constructor(mcpManager: MCPManager, options?: ResourceManagerOptions) {
```

Comment:
change this from options to config and remove the ResourceManagerOptions interface which is an  unnecessary layer of indirection for just one argument